tests/integration/test_integration.py

```diff
@@ -250,6 +291,40 @@ def test_single_trace_too_large(encoding, monkeypatch):
     monkeypatch.setenv("DD_TRACE_API_VERSION", encoding)
 
     t = Tracer()
+    assert t._partial_flush_enabled is True
+    with mock.patch("ddtrace.internal.writer.log") as log:
+        with t.trace("huge"):
+            for i in range(200000):
+                with t.trace("operation") as s:
+                    s.set_tag("a" * 10, "b" * 10)
+        t.shutdown()
+
+        try:
+            log.warning.assert_not_called()
+        except AssertionError:
+            calls = [
+                mock.call(
+                    "trace (%db) larger than payload buffer item limit (%db), dropping",
+                    AnyInt(),
+                    AnyInt(),
+                    AnyInt(),
+                    AnyInt(),
+                )
+            ]
+            print(calls)
+            log.warning.assert_has_calls(calls)
+        else:
+            log.error.assert_not_called()
```

Comment:
This should fix this test. Then I think we're good to go
```suggestion
        calls = [
            mock.call(
                "trace buffer (%s traces %db/%db) cannot fit trace of size %db, dropping",
                AnyInt(),
                AnyInt(),
                AnyInt(),
                AnyInt(),
            )
        ]
        log.warning.assert_has_calls(calls)
        log.error.assert_not_called()
```

Reply:
Tried the above but it doesn't seem like it works because there will be behaviors where some python versions won't have any calls so the ```assert_has_calls``` will then throw an error. It seems like overalls, when partial flushing is enabled, some python versions will have calls while other don't and likewise, for the ones that do have calls, we don't know how many they might have.